src/SoftGlobalScope.jl

```diff
@@ -152,8 +152,21 @@ else
             return Expr(ex.head, (_softscope.(ex.args, Ref(globals), Ref(locals), insertglobal))...)
         elseif isexpr(ex, :kw)
             return Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal))
-        elseif insertglobal && ex.head in assignments && ex.args[1] in globals && !(ex.args[1] in locals)
-            return Expr(:global, Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal)))
+        elseif insertglobal && ex.head in assignments 
+            if ex.args[1] in globals && !(ex.args[1] in locals)
+                # Simple assignment
+                return Expr(:global, Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal)))
+            elseif isexpr(ex.args[1], :tuple)
+                # Assignment to a tuple
+                vars = [var for var in localvars(ex.args[1].args) if (var in globals) && !(var in locals)]
+                if length(vars) > 0
+                    return Expr(:block, Expr(:global, vars...), Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal)))
+                else
+                    return Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal))
```

Comment:
It seems like both of these should use `Expr(ex.head, _softscope.(ex.args, globals, locals, insertglobal)...)`.  Otherwise it will fail on `(a[i+=1], x) = (4,5)` where `i` is a global.

Reply:
I updated it.